skt/reporter.py

```diff
@@ -334,12 +331,41 @@ def getmergeinfo(self):
 
         return result
 
-    def get_kernel_config(self, suffix=None):
-        cfgname = "config.gz" if not suffix else "config_{}.gz".format(suffix)
+    def attach_kernel_config(self, suffix=None):
+        """Add the kernel config to the list of attachments.
+
+        Args:
+            suffix: An optional filename suffix to add to the config file
+                    name. This is helpful for reporting config files for
+                    different kernel architectures.
+
+        Returns:
+            Returns a list of message lines describing the configuration file
+            that is attached. If no config file is found, an empty list is
+            returned.
+
+        """
+        attachment_filename = "config.gz"
+
+        if suffix:
+            attachment_filename = "config_{}.gz".format(suffix)
+
+        # Read the kernel config file, if present. The config file may not be
+        # present if we have completed a merge but we have not yet started the
+        # kernel compile.
+        config_filename = "{}/.config".format('workdir')
+        if os.path.isfile(config_filename):
+            with open(config_filename, 'r') as fileh:
```

Comment:
I would strongly prefer that we stopped doing this: checking the file exists and then opening it. While in the overall scheme, right now, this is insignificant, as we're full of race conditions, this is a bad pattern that needs to be avoided in general. Just open the file, catch the exception, and handle ENOENT.

This is not a requirement for my ACK at this moment, though.

Reply:
Fixing.